legacy/src/main/java/org/opensearch/sql/legacy/cursor/DefaultCursor.java

```diff
@@ -70,42 +95,105 @@ public class DefaultCursor implements Cursor {
   /** To get next batch of result */
   private String scrollId;
 
+  /** To get Point In Time */
+  private String pitId;
+
+  /** To get next batch of result with search after api */
+  public SearchSourceBuilder searchSourceBuilder;
+
+  /** To get last sort values * */
+  private Object[] sortFields;
+
   /** To reduce the number of rows left by fetchSize */
   @NonNull private Integer fetchSize;
 
   private Integer limit;
 
+  /**
+   * {@link NamedXContentRegistry} from {@link SearchModule} used for construct {@link QueryBuilder}
+   * from DSL query string.
+   */
+  private static final NamedXContentRegistry xContentRegistry =
+      new NamedXContentRegistry(
+          new SearchModule(Settings.builder().build(), new ArrayList<>()).getNamedXContents());
```

Comment:
There should be Settings.EMPTY?
Also use Collections.emptyList().

Reply:
Updated to use `Collections.emptyList()`. But I'll be hesitant to change this to `Settings.EMPTY` as I'm not entirely sure if it is same as `Settings.builder().build()` because here we might be trying to retrieve the existing settings and it might not be empty!?

Reply:
``
Debug and see if there is anything in settings. Nothing is being used in Settings...we are creating dummy search module just to use NamedXContents.
``

Reply:
That's right! I see both the values are empty actually.. changes it to Settings.Empty.